.github/workflows/pr_title.yml

```diff
@@ -0,0 +1,22 @@
+name: 'PR Title Check'
+on:
+  pull_request:
+    # check when PR
+    # * is created,
+    # * title is edited, and
+    # * new commits are added (to ensure failing title blocks merging)
+    types: [ opened, reopened, edited, synchronize ]
+
+jobs:
+  title-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: naveenk1223/action-pr-title@master
+        with:
+          #
+          # pull request titles format rules here:
+          # https://submitty.org/developer/how_to_contribute#how-to-make-a-pull-request-pr-to-submitty
+          #
+          # [<TYPE>:<MODULE>] <SUBJECT>
+          #
+          regex: '^\[(Bugfix|Feature|Refactor|Testing|Documentation|VPAT|UI/UX|Dependency):(Submission|Autograding|Forum|Notifications|TAGrading|InstructorUI|RainbowGrades|System|Developer|API)\] .{2,40}$'
```

Comment:
I believe we need to start the regex with `(?:\[SYSADMIN ACTION\] )?`, otherwise looks good. May also want to add `DevDependency` as another type and `PHP` and `JS` as subtypes.

I can talk to @shailpatels on setting up an action for auto renaming dependabot PRs to fit this scheme so that those PRs won't start showing up as failed after this is merged.

Reply:
Sounds good, 
Looks like we can also configure dependabot to change it's commit message/pr title by having a config yaml for it

https://github.com/dependabot/dependabot-core/pull/1227

Reply:
Unfortunately, I don't think dependabot gives us quite enough granularity to specify the appropriate sub-type (PHP / JS), and so with the `prefix` and `prefix_development` config variables we would only be able to get `[Dependency]` or `[DevDependency]`.

Reply:
If I understand correctly the limitations...
I've modified the regex to not require (or allow) the ":module" for dependency or devdependency

Reply:
Yeah, I think losing the subtype is probably not that big a deal for `[Dependency]` or `[DevDependency]` as I do not think anyone was really making use of that information. If someone clamors for it, we can look to re-add it then.